operator/apis/loki/v1/lokistack_types.go

```diff
@@ -791,6 +791,140 @@ type IngestionLimitSpec struct {
 	PerStreamRateLimitBurst int32 `json:"perStreamRateLimitBurst,omitempty"`
 }
 
+// OTLPAttributeAction defines the action to executed when indexing
+// OTLP resource attributes. Resource attributes can be either added
+// to the index, the chunk structured metadata or entirely dropped.
+type OTLPAttributeAction string
+
+const (
+	// OTLPAttributeActionIndexLabel stores a resource attribute as a label, which is part of the index identifying streams.
+	OTLPAttributeActionIndexLabel OTLPAttributeAction = "index_label"
+	// OTLPAttributeActionStructuredMetadata stores an attribute as structured metadata with each log entry.
+	OTLPAttributeActionStructuredMetadata OTLPAttributeAction = "structured_metadata"
```

Comment:
Noticed yesterday, but forgot to add the comment... :roll_eyes: 

Do we want to follow any existing convention with the enum values? Official Kubernetes is `CamelCase`, our convention in Loki Operator so far seems to be `camelCase` and not `snake_case` like in the Loki configuration.

```suggestion
	OTLPAttributeActionIndexLabel OTLPAttributeAction = "indexLabel"
	// OTLPAttributeActionStructuredMetadata stores an attribute as structured metadata with each log entry.
	OTLPAttributeActionStructuredMetadata OTLPAttributeAction = "structuredMetadata"
```

(needs to also update the validations if changed)